### PR TITLE
Mock SDS using FastAPI

### PR DESCRIPTION
**Pull request by MebinAbraham**
### What is the context of this PR?
- Updated the mock SDS API in runner to use FastAPI.
- Enforces that `identifier` must be passed when calling unit data and that it is a valid UUID.
- Dockerised the app
- Pushes the images to Docker Hub, so that runner can pull it. https://hub.docker.com/r/onsdigital/eq-runner-mock-sds/tags
- When making changes, I initially broke the app which went unnoticed until deployment, therefore to reduce this, I've added some **very basic** tests.

### How to review
Using the README, ensure the mock API works as expected both locally and via docker.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Use the official Python 3.9 image as the base image
```
Comment (MebinAbraham) [inline Dockerfile]:
```suggestion
# Use the official Python 3.11 image as the base image
```

File: app/main.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    if filename := guid_filename_map.get(str(dataset_id)):
+        return encrypt_mock_data(load_mock_data(f"mock_data/{filename}.json"))
+
+    raise HTTPException(status_code=404)
+
+
+@app.get("/v1/dataset_metadata")
+def get_sds_dataset_ids(survey_id: str = Query()) -> list[dict]:
```
Comment (katie-gardner) [inline app/main.py]: I think this one needs `period_id` as well (as per you noticing the missing value in the main one)

**Review by petechd — APPROVED** [review]
I've checked all the scripts after installing latest poetry and correct version of python, build and deployed locally, dockerised locally and in pipeline, used dockerised/non dockerised version in local sds launcher and runner, it all seems to work.

File: app/main.py (hunk 2 above)
Comment (katie-gardner) [reply app/main.py]: just for understanding, why add `= Query()`? just for consistency with the other parameters?

**Review by katie-gardner — APPROVED** [review]
Looks great, good setup, pipeline, worked with all the other parts of the pipeline test. One comment which could just be addressed when its updated to be inline with the incoming changes

File: app/main.py (hunk 2 above)
Comment (MebinAbraham) [reply app/main.py]: > I think this one needs period_id as well (as per you noticing the missing value in the main one)

Done.

> just for understanding, why add = Query()? just for consistency with the other parameters?

Kept this but removed the dataset_id one as I am using UUID type now. Rest of the query one have validation including this one now.

Comment (katie-gardner) [reply app/main.py]: Yep makes sense for the validation ones 👍 nice

Comment (katie-gardner) [reply app/main.py]: Will resolve the comment when I have permissions to do so 😄 or feel free

**Review by katie-gardner — APPROVED** [review]

**Review by petechd — APPROVED** [review]
Populates in runner fine, works as expected!

**Pull request merged**